DOCUMENTATION.md

```diff
@@ -0,0 +1,196 @@
+For example, target from your_make.sample:
+```
+deploy/s3/fire_stations: data/out/fire_stations/fire_stations_h3_r8_count.gpkg.gz | deploy/s3 ## deploy/s3 is a dependency from geocint-runner Makefile
+    aws s3 cp data/out/fire_stations/fire_stations_h3_r8_count.gpkg.gz \
+   	 s3://your_s3_bucket/fire_stations_h3_r8_count.gpkg.gz \
+   	 --profile your_s3_profile \
+   	 --acl public-read
+    touch $@
+```
+As a result of this target execution, we don't have a file, which means we have to use the touch $@ line at the end. This command will create an empty deploy/s3/fire_stations file whose creation timestamp will be used by Make-profiler.
+
+if you want to rebuild some targets chain each time please see How geocint pipeline workyou run the pipeline, you must add the initial target of this chain to the TARGETS_TO_CLEAN variable in the configuration file config.inc.sh).
+For example, you have a target’s chain like a (here is a simple chain of targets, just target’s name):
+```
+data/in/some.tiff: | data/in ## initial target, download input data
+data/mid/data_tiff.csv: data/in/some.tiff | data/mid## extract data from raster to intermediate file
+db/table/table_from_csv: data/mid/data_tiff.csv | db/table ## load data from CSV to database
+data/out/output.geojson.gz: db/table/table_from_csv | data/out ## extract data from database to output file
+```
+If you want to rebuild this chain of targets each time when you run the pipeline, you should add data/in/some.tiff target to the TARGETS_TO_CLEAN variable. It means that before each run make will remove data/in/some.tiff if exists and rebuild them and all targets, that have it in list of dependencies (will check dependencies recursively).
+If you do not want to rebuild this chain, just make sure that the final target from your chain doesn’t depend on targets from the TARGETS_TO_CLEAN variable. This method can be useful when you have a large dataset that has not been updated recently and requires heavy and time-consuming pre-processing. In such a case, you probably want to prepare the data and load it into the database once and not repeat this work without changing the input data.
+But! Keep in mind, that dependencies inherited recursively, even if you don’t put data/in/some.tiff in the data/out/output.geojson.gz dependencies list, them will be data/out/output.geojson.gz’s dependency. This means that these two lines are equivalent:
+```
+data/out/output.geojson.gz: db/table/table_from_csv
+data/out/output.geojson.gz: data/in/some.tiff db/table/table_from_csv db/table/table_from_csv | data/out
+```
+You can read more about target's dependencies(prerequisites) in the official [GNU Make manual](https://www.gnu.org/software/make/manual/make.html#Prerequisite-Types)
+
+### User schemas
+
+User schemas can be used for separate pipeline and dev data.
```

Comment:
I do not think that mapaction need this part at all.

Reply:
we should discuss it